package.json

```
@@ -13,34 +13,59 @@
   "unpkg": "dist/ionicons.js",
   "collection": "dist/collection/collection-manifest.json",
   "collection:main": "dist/collection/index.js",
+  "exports": {
+    ".": {
+      "types": "./dist/types/index.d.ts",
+      "default": "./dist/index.js",
+      "require": "./dist/index.cjs.js"
+    },
+    "./components": {
+      "types": "./dist/components/index.d.ts",
+      "import": "./dist/components/index.js"
+    },
+    "./icons": {
+      "types": "./icons/index.d.ts",
+      "import": "./icons/index.mjs",
+      "require": "./icons/index.js"
+    }
+  },
   "scripts": {
-    "compile.scripts": "tsc -p scripts/tsconfig.json",
-    "build": "npm run compile.scripts && npm run build.files && npm run build.component && npm run collection.copy",
-    "build.files": "node scripts/build.js",
+    "build": "run-s clean build.* prettier",
+    "build.files": "tsx scripts/build.js",
```

Should this have the `ts` extension instead?

```suggestion
    "build.files": "tsx scripts/build.ts",
```